index.bs

```diff
@@ -124,33 +104,26 @@ CompressionStream includes GenericTransformStream;
 
 A {{CompressionStream}} has an associated <dfn for=CompressionStream>format</dfn> and <a>compression context</a> <dfn for=CompressionStream>context</dfn>.
 
-The {{CompressionStream}}(*format*) constructor, when invoked, must run these steps:
-    1. If *format* is unsupported in CompressionStream, then throw a TypeError.
-    1. Let *cs* be a new CompressionStream object.
-    1. Set *cs*'s <a for=CompressionStream>format</a> to *format*.
-    1. Let *startAlgorithm* be an algorithm that takes no arguments and returns nothing.
-    1. Let *transformAlgorithm* be an algorithm which takes a *chunk* argument and runs the <a>compress and enqueue a chunk</a> algorithm with *cs* and *chunk*.
-    1. Let *flushAlgorithm* be an algorithm which takes no argument and runs the <a>compress flush and enqueue</a> algorithm with *cs*.
-    1. Let *transform* be the result of calling <a abstract-op>CreateTransformStream</a>(*startAlgorithm*, *transformAlgorithm*, *flushAlgorithm*).
-    1. Set *cs*'s <a>transform</a> to *transform*.
-    1. Return *cs*.
-
-The <dfn>compress and enqueue a chunk</dfn> algorithm, given a CompressionStream object *cs* and a *chunk*, runs these steps:
-    1. If *chunk* is not a {{BufferSource}} type, then return <a>a promise rejected with</a> a TypeError.
+The <dfn constructor for=CompressionStream lt="CompressionStream(format)"><code>new CompressionStream(|format|)</code></dfn> steps are:
+    1. If *format* is unsupported in {{CompressionStream}}, then throw a {{TypeError}}.
+    1. Set [=this=]'s <a for=CompressionStream>format</a> to *format*.
+    1. Let *transformAlgorithm* be an algorithm which takes a *chunk* argument and runs the <a>compress and enqueue a chunk</a> algorithm with [=this=] and *chunk*.
+    1. Let *flushAlgorithm* be an algorithm which takes no argument and runs the <a>compress flush and enqueue</a> algorithm with [=this=].
+    1. Set [=this=]'s [=GenericCompressionStream/transform=] to the result of [=TransformStream/creating=] a {{TransformStream}} with <a for=TransformStream/create><var ignore>transformAlgorithm</var></a> set to *transformAlgorithm* and <a for=TransformStream/create><var ignore>flushAlgorithm</var></a> set to *flushAlgorithm*.
+
+The <dfn>compress and enqueue a chunk</dfn> algorithm, given a {{CompressionStream}} object *cs* and a *chunk*, runs these steps:
+    1. If *chunk* is not a {{BufferSource}} type, then throw a {{TypeError}}.
```

Comment:
I don't think the streams standard converts exceptions to rejections in algorithms.

Reply:
The new wrappers do, specifically for TransformStreams, because all the ones on the web platform are synchronous, and I wanted to make it easy. It's not super-consistent though, e.g. for writable streams I made writeAlgorithm and closeAlgorithm return promises because that's what NativeFS does... Probably I should be more uniform.

Reply:
Oh, I missed that. It would be nice to be more uniform, but I'd rather avoid adding more microtasks if possible.

Reply:
Since this is all about spec-land convenience, I can just branch on the return type :)

Reply:
I didn't end up making this uniform on the Streams side, oops. Something to follow up on perhaps, especially if anyone else wants to make more complicated transform streams in the future.